core/src/main/java/com/amplifyframework/logging/LoggingCategory.java

```diff
@@ -53,15 +56,23 @@ public CategoryType getCategoryType() {
     @NonNull
     @Override
     public Logger forNamespace(@Nullable String namespace) {
-        return getLoggingPlugin().forNamespace(namespace);
+        List<Logger> delegates = new ArrayList<>();
+        boolean defaultPluginAdded = false;
```

Comment:
If you use a `Set`, you might be able to avoid the need to record this state boolean. You could add a plugin twice, and the Set should be able to ignore the duplicate, on the basis of its `hashCode()` value.

Reply:
Using a Set does not work as Loggers of the same type and namespace as well as LoggingPlugins of the same type have different hash codes (these classes don't override the hashCode method and I think the default one only guarantees that the hash codes for two objects will be equal if they represent the same instance).

Reply:
Ah yes true. But, _you could_ add the `equals` and `hashCode` implementations! I think they didn't exist before, since there was only ever one of each, and they didn't need to be "collection safe."

Reply:
Are you suggesting adding equals and hashCode implementations for Logger or LoggingPlugin?